src/RedisResourceManager.php

```diff
@@ -279,6 +295,36 @@ protected function extractPassword($resource, $serverUri)
         return $server['pass'] ?? null;
     }
 
+    /**
+     * Extract password to be used on connection
+     *
+     * @param mixed $resource
+     * @param mixed $serverUri
+     * @return string|null
+     */
+    protected function extractUser($resource, $serverUri)
+    {
+        if (! empty($resource['user'])) {
```

Comment:
I wonder if we can improve the codebase a bit. Since we are expecting `resource` to be an array, we should require `array` in the method signature.
The method is called explicitly and thus we can already do an `is_array()` check beforehand and/or add a proper array shape annotation to the `$resource` storage which reflects the actual type, such as: `array{user?:string,...}`

I can prepare a patch at some point to show what I mean but in case you know what I am talking about, feel free to improve some types on your own.